test/e2e/testdata/artifactgc/artgc-has-gc-but-failed-no-artifacts.yaml

```
@@ -0,0 +1,30 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: artifact-gc-but-failed-no-artifacts-
+spec:
+  entrypoint: main
+  artifactGC:
+    strategy: OnWorkflowDeletion
+  templates:
+    - name: main
+      container:
+        image: argoproj/argosay:v2
+        command:
+          - sh
+          - -c
+        args:
+          - |
+            exit 1
```

I'd probably also `echo` a debug message for clarity and consistency with [other](https://github.com/argoproj/argo-workflows/blob/62e33abded6765568781c519ee3e78b823a99acc/test/e2e/testdata/artifactgc/artgc-artifact-not-written.yaml#L39) [tests](https://github.com/argoproj/argo-workflows/blob/ddbb3c7ad5b498d50514b3c1158ded56e333d75b/examples/continue-on-fail.yaml#L34)
```suggestion
            echo "intentionally not writing anything to disk and intentional failure"
            exit 1
```